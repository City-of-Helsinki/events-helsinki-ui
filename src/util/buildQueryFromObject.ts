/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
import forEach from 'lodash/forEach';
import isArray from 'lodash/isArray';
import isEmpty from 'lodash/isEmpty';
import isNil from 'lodash/isNil';
import isNumber from 'lodash/isNumber';

// eslint-disable-next-line @typescript-eslint/no-explicit-any
const buildQueryFromObject = (obj: { [key: string]: any }) => {
  const query: string[] = [];

  forEach(obj, (filter, key) => {
    if (!isEmpty(filter) || isNumber(filter) || typeof filter === 'boolean') {
      if (isArray(filter)) {
        const items: Array<string | number> = [];

        forEach(filter, (item: string | number) => {
          items.push(encodeURIComponent(item));
        });

        query.push(`${key}=${items.join(',')}`);
      } else if (!isNil(filter)) {
        query.push(`${key}=${encodeURIComponent(filter)}`);
      }
    }
  });

  return query.length ? `?${query.join('&')}` : '';
};

export default buildQueryFromObject;
