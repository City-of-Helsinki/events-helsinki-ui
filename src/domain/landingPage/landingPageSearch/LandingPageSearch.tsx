import { Button, IconSearch, IconTree } from "hds-react";
import React, { FunctionComponent } from "react";
import { useTranslation } from "react-i18next";
import { useHistory } from "react-router";

import CategoryFilter from "../../../common/components/category/CategoryFilter";
import DateSelector from "../../../common/components/dateSelector/DateSelector";
import Link from "../../../common/components/link/Link";
import MobileDateSelector from "../../../common/components/mobileDateSelector/MobileDateSelector";
import SearchAutosuggest from "../../../common/components/search/SearchAutosuggest";
import SearchLabel from "../../../common/components/search/searchLabel/SearchLabel";
import { AutosuggestMenuOption, Category } from "../../../common/types";
import { CATEGORIES } from "../../../constants";
import useLocale from "../../../hooks/useLocale";
import IconCultureAndArts from "../../../icons/IconCultureAndArts";
import IconDance from "../../../icons/IconDance";
import IconFood from "../../../icons/IconFood";
import IconMovies from "../../../icons/IconMovies";
import IconMuseum from "../../../icons/IconMuseum";
import IconMusic from "../../../icons/IconMusic";
import IconParticipate from "../../../icons/IconParticipate";
import IconSports from "../../../icons/IconSports";
import IconTheatre from "../../../icons/IconTheatre";
import { getSearchQuery } from "../../../util/searchUtils";
import styles from "./landingPageSearch.module.scss";

const Search: FunctionComponent = () => {
  const { t } = useTranslation();
  const locale = useLocale();
  const [dateTypes, setDateTypes] = React.useState<string[]>([]);
  const [startDate, setStartDate] = React.useState<Date | null>(null);
  const [endDate, setEndDate] = React.useState<Date | null>(null);
  const [isCustomDate, setIsCustomDate] = React.useState<boolean>(false);
  const [searchValue, setSearchValue] = React.useState("");
  const { push } = useHistory();

  const handleCategoryClick = (category: Category) => {
    const search = getSearchQuery({
      categories: [category.value],
      dateTypes,
      districts: [],
      endDate,
      isFree: false,
      keywordNot: [],
      keywords: [],
      places: [],
      publisher: null,
      search: "",
      startDate,
      targets: []
    });

    push({ pathname: `/${locale}/events`, search });
  };

  const categories = React.useMemo(() => {
    return [
      {
        icon: <IconMovies />,
        text: t("home.category.movie"),
        value: CATEGORIES.MOVIE
      },
      {
        icon: <IconMusic />,
        text: t("home.category.music"),
        value: CATEGORIES.MUSIC
      },
      {
        icon: <IconSports />,
        text: t("home.category.sport"),
        value: CATEGORIES.SPORT
      },
      {
        icon: <IconMuseum />,
        text: t("home.category.museum"),
        value: CATEGORIES.MUSEUM
      },
      {
        icon: <IconDance />,
        text: t("home.category.dance"),
        value: CATEGORIES.DANCE
      },
      {
        icon: <IconCultureAndArts />,
        text: t("home.category.culture"),
        value: CATEGORIES.CULTURE
      },
      {
        icon: <IconTree />,
        text: t("home.category.nature"),
        value: CATEGORIES.NATURE
      },
      {
        icon: <IconParticipate />,
        text: t("home.category.influence"),
        value: CATEGORIES.INFLUENCE
      },
      {
        icon: <IconTheatre />,
        text: t("home.category.theatre"),
        value: CATEGORIES.THEATRE
      },
      {
        className: styles.categoryFood,
        icon: <IconFood />,
        text: t("home.category.food"),
        value: CATEGORIES.FOOD
      }
    ];
  }, [t]);

  const handleChangeDateTypes = (value: string[]) => {
    setDateTypes(value);
  };

  const toggleIsCustomDate = () => {
    setIsCustomDate(!isCustomDate);
  };

  const moveToSearchPage = React.useCallback(() => {
    const search = getSearchQuery({
      categories: [],
      dateTypes,
      districts: [],
      endDate,
      isFree: false,
      keywordNot: [],
      keywords: [],
      places: [],
      publisher: null,
      search: searchValue,
      startDate,
      targets: []
    });

    push({ pathname: `/${locale}/events`, search });
  }, [dateTypes, endDate, locale, push, searchValue, startDate]);

  const handleMenuOptionClick = (option: AutosuggestMenuOption) => {
    const type = option.type;
    const value = option.value;
    const searchValue = option.type === "search" ? option.text : "";

    const search = getSearchQuery({
      categories: [],
      dateTypes,
      districts: [],
      endDate,
      isFree: false,
      keywordNot: [],
      keywords: type === "keyword" || type === "yso" ? [value] : [],
      places: [],
      publisher: null,
      search: searchValue,
      startDate,
      targets: []
    });

    push({ pathname: `/${locale}/events`, search });
  };

  return (
    <>
      <div className={styles.landingPageSearch}>
        {/* Hide Suprise me button on MVP version */}
        {/* <SupriseMeButton onClick={handleClickSupriseMe} /> */}
        <div className={styles.searchRow}>
          <div className={styles.titleWrapper}>
            <h2>{t("home.search.title")}</h2>
          </div>
          <div className={styles.autosuggestWrapper}>
            <SearchLabel htmlFor={"search"}>
              {t("home.search.labelSearchField")}
            </SearchLabel>
            <SearchAutosuggest
              categories={[]}
              name="search"
              onChangeSearchValue={setSearchValue}
              onOptionClick={handleMenuOptionClick}
              placeholder={t("home.search.placeholder")}
              searchValue={searchValue}
            />
          </div>
          <div className={styles.dateAndButtonWrapper}>
            <div className={styles.dateSelectorWrapper}>
              <SearchLabel color="black" htmlFor="date" srOnly={true}>
                {t("home.search.labelDateRange")}
              </SearchLabel>
              <div className={styles.desktopDateSelector}>
                <DateSelector
                  dateTypes={dateTypes}
                  endDate={endDate}
                  isCustomDate={isCustomDate}
                  name="date"
                  onChangeDateTypes={handleChangeDateTypes}
                  onChangeEndDate={setEndDate}
                  onChangeStartDate={setStartDate}
                  startDate={startDate}
                  toggleIsCustomDate={toggleIsCustomDate}
                />
              </div>
              <MobileDateSelector
                dateTypes={dateTypes}
                endDate={endDate}
                name={"mobile_date"}
                onChangeDateTypes={handleChangeDateTypes}
                onChangeEndDate={setEndDate}
                onChangeStartDate={setStartDate}
                startDate={startDate}
              />
            </div>
            <div className={styles.buttonWrapper}>
              <Button
                fullWidth={true}
                iconLeft={<IconSearch />}
                onClick={moveToSearchPage}
                variant="success"
              >
                {t("home.search.buttonSearch")}
              </Button>
            </div>
          </div>
        </div>
        <div className={styles.linkRow}>
          <Link color="white" size="small" to={`/${locale}/events`}>
            {t("home.search.linkAdvancedSearch")}
          </Link>
        </div>
      </div>
      <div className={styles.categoriesWrapper}>
        {categories.map(category => {
          return (
            <CategoryFilter
              className={category.className}
              key={category.value}
              icon={category.icon}
              onClick={handleCategoryClick}
              text={category.text}
              value={category.value}
            />
          );
        })}
      </div>
    </>
  );
};

export default Search;
