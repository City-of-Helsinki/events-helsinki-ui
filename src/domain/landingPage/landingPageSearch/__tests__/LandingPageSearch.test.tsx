import { advanceTo } from 'jest-date-mock';
import React from 'react';

import { KeywordListDocument } from '../../../../generated/graphql';
import { fakeKeywords } from '../../../../util/mockDataUtils';
import {
  actWait,
  configure,
  render,
  screen,
  userEvent,
} from '../../../../util/testUtils';
import LandingPageSearch from '../LandingPageSearch';

configure({ defaultHidden: true });

const searchValue = 'jaz';
const keywords = fakeKeywords(2, [
  { name: { fi: 'Jazz' } },
  { name: { fi: 'musiikkiklubit' } },
]);
const keywordsResponse = { data: { keywordList: keywords } };

const mocks = [
  {
    request: {
      query: KeywordListDocument,
      variables: {
        hasUpcomingEvents: true,
        pageSize: 5,
        text: searchValue,
      },
    },
    result: keywordsResponse,
  },
];

test('should route to event search page after clicking submit button', async () => {
  const { history } = render(<LandingPageSearch />, { mocks });

  userEvent.click(screen.getByRole('button', { name: /hae/i }));
  expect(history.location.pathname).toBe('/fi/events');
  expect(history.location.search).toBe(``);
});

test('should route to event search page with correct search query after clicking submit button', async () => {
  const { history } = render(<LandingPageSearch />, { mocks });

  const searchInput = screen.getByRole('textbox', { name: /mitä etsit\?/i });
  userEvent.type(searchInput, searchValue);

  // Check that auto-suggest menu is open
  expect(screen.getByText(/hakuehdotuksia/i)).toBeInTheDocument();

  userEvent.click(screen.getByRole('button', { name: /hae/i }));
  expect(history.location.pathname).toBe('/fi/events');
  expect(history.location.search).toBe(`?text=${searchValue}`);
});

test('should route to event search page after clicking autosuggest menu item', async () => {
  const { history } = render(<LandingPageSearch />, { mocks });

  const searchInput = screen.getByRole('textbox', { name: /mitä etsit\?/i });
  userEvent.type(searchInput, searchValue);

  // Wait autosuggest search internal input debounce
  await actWait(500);

  userEvent.click(screen.getByRole('option', { name: /musiikkiklubit/i }));
  expect(history.location.pathname).toBe('/fi/events');
  expect(history.location.search).toBe(`?text=musiikkiklubit`);
});

test('should route to event search page after clicking category', async () => {
  const { history } = render(<LandingPageSearch />, { mocks });

  userEvent.click(screen.getByText(/kulttuuri ja taide/i));

  expect(history.location.pathname).toBe('/fi/events');
  expect(history.location.search).toBe('?categories=culture');
});

test('should route to event search page after selecting category type and pressing submit button', async () => {
  const { history } = render(<LandingPageSearch />, { mocks });

  userEvent.click(screen.getByRole('button', { name: /valitse ajankohta/i }));
  userEvent.click(screen.getByRole('checkbox', { name: /tänään/i }));

  userEvent.click(screen.getByRole('button', { name: /hae/i }));
  expect(history.location.pathname).toBe('/fi/events');
  expect(history.location.search).toBe('?dateTypes=today');
});

test('should route to event search page after selecting start date and pressing submit button', async () => {
  advanceTo('2020-10-04');
  const { history } = render(<LandingPageSearch />, { mocks });

  userEvent.click(screen.getByRole('button', { name: /valitse ajankohta/i }));
  userEvent.click(
    // The reason to use getAllByRole is that there is also mobile date selector with same text,
    // which is hidden using css
    screen.getAllByRole('button', { name: /valitse päivät/i })[0]
  );
  userEvent.click(screen.getByRole('option', { name: /day\-6/i }));

  userEvent.click(screen.getByRole('button', { name: /hae/i }));
  expect(history.location.pathname).toBe('/fi/events');
  expect(history.location.search).toBe('?start=2020-10-06');
});
