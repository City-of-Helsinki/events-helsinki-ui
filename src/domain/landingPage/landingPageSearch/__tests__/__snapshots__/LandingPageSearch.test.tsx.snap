// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageSearch should match snapshot 1`] = `
<div
  class="landingPageSearch"
>
  <div
    class="searchRow"
  >
    <div
      class="titleWrapper"
    >
      <h2>
        Löydä tekemistä
      </h2>
    </div>
    <div
      class="autosuggestWrapper"
    >
      <label
        class="searchLabel white"
        for="search"
      >
        Mitä etsit?
      </label>
      <div
        class="searchAutosuggest"
      >
        <div
          class="iconWrapper"
        >
          <svg
            class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <g
              fill="none"
              fill-rule="evenodd"
            >
              <path
                d="M0 0h24v24H0z"
              />
              <path
                d="M15 1a8 8 0 11-4.798 14.402l-6.401 6.4-1.591-1.59 6.398-6.4A8 8 0 0115 1zm0 2a6 6 0 100 12 6 6 0 000-12z"
                fill="currentColor"
              />
            </g>
          </svg>
        </div>
        <div
          class="inputWrapper"
        >
          <input
            id="search"
            name="search"
            placeholder="Kirjoita hakusana, esim. rock tai jooga"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="dateAndButtonWrapper"
    >
      <div
        class="dateSelectorWrapper"
      >
        <label
          class="searchLabel black srOnly"
          for="date"
        >
          Mille ajankohdalle etsit?
        </label>
        <div
          class="desktopDateSelector"
        >
          <div
            class="dateSelector"
          >
            <button
              aria-expanded="false"
              aria-haspopup="true"
              aria-label="Valitse ajankohta"
              class="button"
              type="button"
            >
              <div
                class="iconWrapper"
              >
                <svg
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <path
                      d="M0 0h24v24H0z"
                    />
                    <path
                      d="M17 12a6 6 0 110 12 6 6 0 010-12zm0 2a4 4 0 100 8 4 4 0 000-8zm0-12a1 1 0 011 1v1h4l.002 9.103A7.018 7.018 0 0020 11.674L20 11H4v8l6.071.001a6.95 6.95 0 00.603 2L2 21V4h4V3a1 1 0 112 0v1h8V3a1 1 0 011-1zm.5 13v2.94l1.53 1.53-1.06 1.06L16 18.56V15h1.5zM20 6H4v3h16V6z"
                      fill="currentColor"
                    />
                  </g>
                </svg>
              </div>
              <div
                class="info"
              >
                <div
                  class="buttonTextWrapper"
                >
                  Valitse ajankohta
                </div>
              </div>
              <div
                class="arrowWrapper"
              >
                <svg
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <path
                      d="M0 0h24v24H0z"
                    />
                    <path
                      d="M12 13.5l5-5 1.5 1.5-6.5 6.5L5.5 10 7 8.5z"
                      fill="currentColor"
                    />
                  </g>
                </svg>
              </div>
            </button>
          </div>
        </div>
        <div
          class="mobileDateSelector"
        >
          <button
            aria-pressed="false"
            class="toggleButton"
            type="button"
          >
            <div
              class="innerWrapper"
            >
              <span>
                Tänään
              </span>
            </div>
          </button>
          <button
            aria-pressed="false"
            class="toggleButton"
            type="button"
          >
            <div
              class="innerWrapper"
            >
              <span>
                Huomenna
              </span>
            </div>
          </button>
          <button
            aria-pressed="false"
            class="toggleButton"
            type="button"
          >
            <div
              class="innerWrapper"
            >
              <span>
                Tällä viikolla
              </span>
            </div>
          </button>
          <button
            aria-pressed="false"
            class="toggleButton"
            type="button"
          >
            <div
              class="innerWrapper"
            >
              <span>
                Viikonloppuna
              </span>
            </div>
          </button>
          <div
            class="menuWrapper"
          >
            <button
              aria-pressed="false"
              class="toggleButton"
              data-testid="open-date-selector-button"
              type="button"
            >
              <div
                class="innerWrapper"
              >
                <svg
                  class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik iconCalendar"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g
                    fill="none"
                    fill-rule="evenodd"
                  >
                    <path
                      d="M0 0h24v24H0z"
                    />
                    <path
                      d="M17 2a1 1 0 011 1v1h4v17H2V4h4V3a1 1 0 112 0v1h8V3a1 1 0 011-1zm3 9H4v8h16v-8zm0-5H4v3h16V6z"
                      fill="currentColor"
                    />
                  </g>
                </svg>
                <span>
                  Valitse päivät
                </span>
              </div>
            </button>
            <div
              class="mobileDateSelectorMenu"
              data-testid="mobile-date-selector-menu"
            >
              <div
                class="wrapper"
              >
                <div
                  class="react-datepicker-wrapper"
                >
                  <div
                    class="react-datepicker__input-container"
                  >
                    <div
                      class="dateRangeInputsContainer"
                    >
                      <div
                        class="dateInputWrapper"
                      >
                        <label
                          class="label"
                          for="mobile_date_start_date"
                        >
                          Alkamispäivä
                        </label>
                        <div
                          class="formatInfo"
                        >
                          pp.kk.vvvv
                        </div>
                        <div
                          class="inputWrapper"
                        >
                          <div
                            class="input"
                          >
                            <input
                              class=""
                              id="mobile_date_start_date"
                              name="mobile_date_start_date"
                              value=""
                            />
                          </div>
                          <div
                            class="icon"
                          >
                            <svg
                              class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                              viewBox="0 0 24 24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <g
                                fill="none"
                                fill-rule="evenodd"
                              >
                                <path
                                  d="M0 0h24v24H0z"
                                />
                                <path
                                  d="M20 14v3h3v2h-3v3h-2v-3h-3v-2h3v-3h2zM17 2a1 1 0 011 1v1h4v8h-2v-1H4v8h9v2H2V4h4V3a1 1 0 112 0v1h8V3a1 1 0 011-1zm3 4H4v3h16V6z"
                                  fill="currentColor"
                                />
                              </g>
                            </svg>
                          </div>
                        </div>
                      </div>
                      <div
                        class="dateSeparator"
                      >
                        —
                      </div>
                      <div
                        class="dateInputWrapper"
                      >
                        <label
                          for="mobile_date_end_date"
                        >
                          Loppumispäivä
                        </label>
                        <div
                          class="formatInfo"
                        >
                          pp.kk.vvvv
                        </div>
                        <div
                          class="inputWrapper"
                        >
                          <div
                            class="input"
                          >
                            <input
                              class=""
                              id="mobile_date_end_date"
                              name="mobile_date_end_date"
                              value=""
                            />
                          </div>
                          <div
                            class="icon"
                          >
                            <svg
                              class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
                              viewBox="0 0 24 24"
                              xmlns="http://www.w3.org/2000/svg"
                            >
                              <g
                                fill="none"
                                fill-rule="evenodd"
                              >
                                <path
                                  d="M0 0h24v24H0z"
                                />
                                <path
                                  d="M20 14v3h3v2h-3v3h-2v-3h-3v-2h3v-3h2zM17 2a1 1 0 011 1v1h4v8h-2v-1H4v8h9v2H2V4h4V3a1 1 0 112 0v1h8V3a1 1 0 011-1zm3 4H4v3h16V6z"
                                  fill="currentColor"
                                />
                              </g>
                            </svg>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="react-datepicker-popper react-datepicker__no-popper"
                  style="position: absolute; top: 0px; left: 0px; opacity: 0; pointer-events: none;"
                >
                  <div>
                    <div
                      class="react-datepicker"
                    >
                      <div
                        class="react-datepicker__triangle"
                      />
                      <button
                        class="react-datepicker__navigation react-datepicker__navigation--next"
                        type="button"
                      >
                        Next Month
                      </button>
                      <div
                        class="react-datepicker__month-container"
                      >
                        <div
                          class="react-datepicker__header"
                        >
                          <div
                            class="react-datepicker__current-month"
                          >
                            elokuu 2020
                          </div>
                          <div
                            class="react-datepicker__header__dropdown react-datepicker__header__dropdown--scroll"
                          />
                          <div
                            class="react-datepicker__day-names"
                          >
                            <div
                              class="react-datepicker__day-name"
                            >
                              ma
                            </div>
                            <div
                              class="react-datepicker__day-name"
                            >
                              ti
                            </div>
                            <div
                              class="react-datepicker__day-name"
                            >
                              ke
                            </div>
                            <div
                              class="react-datepicker__day-name"
                            >
                              to
                            </div>
                            <div
                              class="react-datepicker__day-name"
                            >
                              pe
                            </div>
                            <div
                              class="react-datepicker__day-name"
                            >
                              la
                            </div>
                            <div
                              class="react-datepicker__day-name"
                            >
                              su
                            </div>
                          </div>
                        </div>
                        <div
                          aria-label="month-2020-08"
                          class="react-datepicker__month"
                          role="listbox"
                        >
                          <div
                            class="react-datepicker__week"
                          >
                            <div
                              aria-disabled="true"
                              aria-label="day-27"
                              class="react-datepicker__day react-datepicker__day--027 react-datepicker__day--disabled react-datepicker__day--outside-month"
                              role="option"
                            >
                              27
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-28"
                              class="react-datepicker__day react-datepicker__day--028 react-datepicker__day--disabled react-datepicker__day--outside-month"
                              role="option"
                            >
                              28
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-29"
                              class="react-datepicker__day react-datepicker__day--029 react-datepicker__day--disabled react-datepicker__day--outside-month"
                              role="option"
                            >
                              29
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-30"
                              class="react-datepicker__day react-datepicker__day--030 react-datepicker__day--disabled react-datepicker__day--outside-month"
                              role="option"
                            >
                              30
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-31"
                              class="react-datepicker__day react-datepicker__day--031 react-datepicker__day--disabled react-datepicker__day--outside-month"
                              role="option"
                            >
                              31
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-1"
                              class="react-datepicker__day react-datepicker__day--001 react-datepicker__day--disabled react-datepicker__day--weekend"
                              role="option"
                            >
                              1
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-2"
                              class="react-datepicker__day react-datepicker__day--002 react-datepicker__day--disabled react-datepicker__day--weekend"
                              role="option"
                            >
                              2
                            </div>
                          </div>
                          <div
                            class="react-datepicker__week"
                          >
                            <div
                              aria-disabled="true"
                              aria-label="day-3"
                              class="react-datepicker__day react-datepicker__day--003 react-datepicker__day--disabled"
                              role="option"
                            >
                              3
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-4"
                              class="react-datepicker__day react-datepicker__day--004 react-datepicker__day--disabled"
                              role="option"
                            >
                              4
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-5"
                              class="react-datepicker__day react-datepicker__day--005 react-datepicker__day--disabled"
                              role="option"
                            >
                              5
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-6"
                              class="react-datepicker__day react-datepicker__day--006 react-datepicker__day--disabled"
                              role="option"
                            >
                              6
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-7"
                              class="react-datepicker__day react-datepicker__day--007 react-datepicker__day--disabled"
                              role="option"
                            >
                              7
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-8"
                              class="react-datepicker__day react-datepicker__day--008 react-datepicker__day--disabled react-datepicker__day--weekend"
                              role="option"
                            >
                              8
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-9"
                              class="react-datepicker__day react-datepicker__day--009 react-datepicker__day--disabled react-datepicker__day--weekend"
                              role="option"
                            >
                              9
                            </div>
                          </div>
                          <div
                            class="react-datepicker__week"
                          >
                            <div
                              aria-disabled="true"
                              aria-label="day-10"
                              class="react-datepicker__day react-datepicker__day--010 react-datepicker__day--disabled"
                              role="option"
                            >
                              10
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-11"
                              class="react-datepicker__day react-datepicker__day--011 react-datepicker__day--disabled"
                              role="option"
                            >
                              11
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-12"
                              class="react-datepicker__day react-datepicker__day--012 react-datepicker__day--disabled"
                              role="option"
                            >
                              12
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-13"
                              class="react-datepicker__day react-datepicker__day--013 react-datepicker__day--disabled"
                              role="option"
                            >
                              13
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-14"
                              class="react-datepicker__day react-datepicker__day--014 react-datepicker__day--disabled"
                              role="option"
                            >
                              14
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-15"
                              class="react-datepicker__day react-datepicker__day--015 react-datepicker__day--disabled react-datepicker__day--weekend"
                              role="option"
                            >
                              15
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-16"
                              class="react-datepicker__day react-datepicker__day--016 react-datepicker__day--disabled react-datepicker__day--weekend"
                              role="option"
                            >
                              16
                            </div>
                          </div>
                          <div
                            class="react-datepicker__week"
                          >
                            <div
                              aria-disabled="true"
                              aria-label="day-17"
                              class="react-datepicker__day react-datepicker__day--017 react-datepicker__day--disabled"
                              role="option"
                            >
                              17
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-18"
                              class="react-datepicker__day react-datepicker__day--018 react-datepicker__day--disabled"
                              role="option"
                            >
                              18
                            </div>
                            <div
                              aria-disabled="true"
                              aria-label="day-19"
                              class="react-datepicker__day react-datepicker__day--019 react-datepicker__day--disabled"
                              role="option"
                            >
                              19
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-20"
                              class="react-datepicker__day react-datepicker__day--020 react-datepicker__day--keyboard-selected react-datepicker__day--today"
                              role="option"
                            >
                              20
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-21"
                              class="react-datepicker__day react-datepicker__day--021"
                              role="option"
                            >
                              21
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-22"
                              class="react-datepicker__day react-datepicker__day--022 react-datepicker__day--weekend"
                              role="option"
                            >
                              22
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-23"
                              class="react-datepicker__day react-datepicker__day--023 react-datepicker__day--weekend"
                              role="option"
                            >
                              23
                            </div>
                          </div>
                          <div
                            class="react-datepicker__week"
                          >
                            <div
                              aria-disabled="false"
                              aria-label="day-24"
                              class="react-datepicker__day react-datepicker__day--024"
                              role="option"
                            >
                              24
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-25"
                              class="react-datepicker__day react-datepicker__day--025"
                              role="option"
                            >
                              25
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-26"
                              class="react-datepicker__day react-datepicker__day--026"
                              role="option"
                            >
                              26
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-27"
                              class="react-datepicker__day react-datepicker__day--027"
                              role="option"
                            >
                              27
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-28"
                              class="react-datepicker__day react-datepicker__day--028"
                              role="option"
                            >
                              28
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-29"
                              class="react-datepicker__day react-datepicker__day--029 react-datepicker__day--weekend"
                              role="option"
                            >
                              29
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-30"
                              class="react-datepicker__day react-datepicker__day--030 react-datepicker__day--weekend"
                              role="option"
                            >
                              30
                            </div>
                          </div>
                          <div
                            class="react-datepicker__week"
                          >
                            <div
                              aria-disabled="false"
                              aria-label="day-31"
                              class="react-datepicker__day react-datepicker__day--031"
                              role="option"
                            >
                              31
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-1"
                              class="react-datepicker__day react-datepicker__day--001 react-datepicker__day--outside-month"
                              role="option"
                            >
                              1
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-2"
                              class="react-datepicker__day react-datepicker__day--002 react-datepicker__day--outside-month"
                              role="option"
                            >
                              2
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-3"
                              class="react-datepicker__day react-datepicker__day--003 react-datepicker__day--outside-month"
                              role="option"
                            >
                              3
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-4"
                              class="react-datepicker__day react-datepicker__day--004 react-datepicker__day--outside-month"
                              role="option"
                            >
                              4
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-5"
                              class="react-datepicker__day react-datepicker__day--005 react-datepicker__day--weekend react-datepicker__day--outside-month"
                              role="option"
                            >
                              5
                            </div>
                            <div
                              aria-disabled="false"
                              aria-label="day-6"
                              class="react-datepicker__day react-datepicker__day--006 react-datepicker__day--weekend react-datepicker__day--outside-month"
                              role="option"
                            >
                              6
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <button
                class="button btnClose"
              >
                <div
                  class="buttonText"
                >
                  Sulje
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="buttonWrapper"
      >
        <button
          class="Button-module_button__3wgee button_hds-button__2A0je Button-module_success__1LdEx button_hds-button--success__9hpuD Button-module_fullWidth__2AR2o button_hds-button--fullwidth__1_hG9"
          type="button"
        >
          <div
            aria-hidden="true"
            class="Button-module_icon__3Wlc6 button_hds-icon__17j8Z"
          >
            <svg
              class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <path
                  d="M0 0h24v24H0z"
                />
                <path
                  d="M15 1a8 8 0 11-4.798 14.402l-6.401 6.4-1.591-1.59 6.398-6.4A8 8 0 0115 1zm0 2a6 6 0 100 12 6 6 0 000-12z"
                  fill="currentColor"
                />
              </g>
            </svg>
          </div>
          <span
            class="Button-module_label__2zMLC button_hds-button__label__2EQa-"
          >
            Hae
          </span>
        </button>
      </div>
    </div>
  </div>
  <div
    class="linkRow"
  >
    <a
      class="link whiteColor smallSize"
      href="/fi/events"
    >
      Tarkennettu haku
      <svg
        class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g
          fill="none"
          fill-rule="evenodd"
        >
          <path
            d="M0 24V0h24v24z"
          />
          <path
            d="M13 12.5l-5-5L9.5 6l6.5 6.5L9.5 19 8 17.5z"
            fill="currentColor"
          />
        </g>
      </svg>
    </a>
  </div>
</div>
`;
